src/Type/Php/RegexArrayShapeMatcher.php

```diff
@@ -372,15 +375,22 @@ private function parseGroups(string $regex): ?array
 	{
 		if (self::$parser === null) {
 			/** @throws void */
-			self::$parser = Llk::load(new Read('hoa://Library/Regex/Grammar.pp'));
+			self::$parser = Llk::load(new Read(__DIR__ . '/../../../resources/RegexGrammar.pp'));
 		}
 
 		try {
-			$ast = self::$parser->parse($regex);
-		} catch (Exception) {
+			Strings::match('', $regex);
+		} catch (RegexpException $e) {
+			// pattern is invalid, so let the RegularExpressionPatternRule report it
 			return null;
 		}
```

Comment:
I just added a first validation pass here @staabm, and a ShouldNotHappenException below so that if the grammar fails we know.. But it's a bit brutal to fail the whole process due to that. Is there not another softer exception than ShouldNotHappenException to surface an internal error to users without aborting the entire run?

Reply:
I think we should have a separate rule which depends on a feature flag .. maybe bleeding edge.. maybe some separate flag

Reply:
Ok I got rid of the throw now.